app/experimenter/experiments/tests/mixins.py

```diff
@@ -5,6 +5,73 @@
 from experimenter.openidc.tests.factories import UserFactory
 
 
+class MockNormandyMixin(object):
+
+    def setUp(self):
+        super().setUp()
+
+        mock_normandy_requests_get_patcher = mock.patch(
+            "experimenter.experiments.normandy.requests.get"
+        )
+        self.mock_normandy_requests_get = (
+            mock_normandy_requests_get_patcher.start()
+        )
+        self.addCleanup(mock_normandy_requests_get_patcher.stop)
+        self.mock_normandy_requests_get.return_value = (
+            self.buildMockSuccessEnabledResponse()
+        )
+
+    def buildMockSuccessEnabledResponse(self):
+        mock_response_data = {
+            "approved_revision": {
+                "enabled": True,
+                "enabled_states": [{"creator": {"email": "dev@example.com"}}],
+            }
+        }
+        mock_response = mock.Mock()
+        mock_response.json = mock.Mock()
+        mock_response.json.return_value = mock_response_data
+        mock_response.raise_for_status = mock.Mock()
+        mock_response.raise_for_status.side_effect = None
+        mock_response.status_code = 200
+        return mock_response
+
+    def buildMockFailedResponse(self):
+        mock_response_data = {"message": "id not found"}
+        mock_response = mock.Mock()
+        mock_response.json = mock.Mock()
+        mock_response.json.return_value = mock_response_data
+        mock_response.raise_for_status = mock.Mock()
+        mock_response.raise_for_status.side_effect = None
+        mock_response.status_code = 404
+        return mock_response
+
+    def buildMockSuccessDisabledResponse(self):
+        mock_response_data = {
+            "approved_revision": {
+                "enabled": False,
+                "enabled_states": [{"creator": {"email": "dev@example.com"}}],
+            }
+        }
+        mock_response = mock.Mock()
+        mock_response.json = mock.Mock()
+        mock_response.json.return_value = mock_response_data
+        mock_response.raise_for_status = mock.Mock()
+        mock_response.raise_for_status.side_effect = None
+        mock_response.status_code = 200
+        return mock_response
+
+    def setUpMockNormandyFailWithSpecifiedID(self, id):
```

Comment:
Tiny nit: `id` is a reserved word in python (it returns a unique identifier for an in memory object)

https://docs.python.org/3/library/functions.html#id

Reply:
okie doke, will change to normandy_id?